src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/discussions/PostsListPage.java

```diff
@@ -129,4 +142,58 @@ public PostsListPage clickUsernameLink() {
   public boolean isUserPageHeaderVisible() {
     return userPageHeader.isDisplayed();
   }
+
+
+  public boolean isUpvoteButtonVisible(int index) {
+    WebElement button = replyUpvoteButton.get(index);
+    wait.forElementVisible(button);
+    return button.isDisplayed();
+  }
+
+  public String getVoteCount(int index) {
+    WebElement voteCountArea = replyVoteCount.get(index);
+    wait.forElementVisible(voteCountArea);
+    return voteCountArea.getText();
+  }
+
+  public void clickUpvoteButton(int postIndex) {
+    WebElement button = replyUpvoteButton.get(postIndex);
+    wait.forElementClickable(button);
+    button.click();
+  }
+
+  public void waitForVoteCountToChange(int postIndex, String voteCount) {
+    WebElement voteArea = replyVoteCount.get(postIndex);
+    wait.forTextNotInElement(voteArea, voteCount);
+  }
+
+  /**
+   * Wait for the noticeable time lag between vote and vote value change to pass
+   */
+  public void waitForVoteCountChangeTimeLagToPass() {
+    try {
+      //This sleep was introduced because of noticeable lag between vote and vote value change
+      Thread.sleep(2000);
+    } catch (InterruptedException e) {
+      e.printStackTrace();
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Use a logger to log this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1148)